README.md
DerivativeClicker
=================

Github Pages hosted version of Derivative Clicker


Changelog
=========

###Version 0.34 (01/05/2015)
Added infini-currency tracking. Currently, these do nothing.

###Version 0.331 (08/17/2014)
Made tooltips less ugly, edited upper section of UI, added achievement number display

###Version 0.33 (08/17/2014)
Added base achievement code and a few achievements, added chart options, fixed bug with costs of click booster

###Version 0.321 (08/05/2014)
Fixed a bug relating to reset currency buyable display, added number suffixes and link to Github repository

###Version 0.32 (08/01/2014)
Add a new option that allows you set the minimum tick length, changed tick length display, fixed clicks to gain going negative, change initial computer price

###Version 0.31 (07/28/2014)
Make tooltips not flash

###Version 0.3 (07/22/2014)
Added reset currency buyables, which have various effects; players start with $0.10 on reset to allow for neverclick runs

###Version 0.222 (07/08/2014)
Number to buy now works with upgrades (except click improver), fixed a minor bug causing money to go negative

###Version 0.221 (07/08/2014)
Tier unlocking makes more sense, fixed negative buildings bug

###Version 0.22 (07/04/2014)
Added 7th tier, changed reset currency calculation, proper upgrade hiding, improved multiplier calculation, nerfed upgrade multipliers, added alternate number view, changed some names, made some things more efficient, rebalancing.

###Version 0.21 (06/27/2014)
Changed mathematician prestige formula to an explicit formula, bugfixes for version 0.2

###Version 0.2 (06/27/2014)
Added a sixth derivative (currently unlocked on first 2nd tier reset) as well as related sixth tier stuff, made 5th tier reset possible, refactored some code.

###Version 0.184 (06/25/2014)
Changed prestige proof formula to an explicit formula, moved update function into global namespace

###Version 0.183 (06/20/2014)
Added number suffixes to trigintillion, added total ticks statistic, added time multiplier (which multiplies everything by 10 when you get to 100 ms tick length, to speed up the game)

###Version 0.182 (06/13/2014)
More number display fixes, buttons use onmousedown events for greater responsiveness

###Version 0.181 (06/06/2014)
Bugfixes: global multiplier gets recalculated when save imports, numbers that displayed incorrectly have been fixed, some incorrect prices are changed, updating should be a little more responsive

###Version 0.18 (05/30/14)
Added tiered resets, and made it way easier to get reset currency.

###Version 0.17 (05/16/14)
Fixed some graphical bugs, made chart not show more than 100 points, extended number suffixes to a vigintillion, fixed serious bug due to Chrome and Firefox locking updates to once per second while in the background.

###Version 0.16 (05/06/14)
Importing and exporting saves is now possible on the Options screen.

###Version 0.15 (05/01/14)
Charts are now more responsive, and there's a button to change the axis.

###Version 0.14 (04/30/14)
A lot of changes to try to speed up the chart. Toggling actually disables the chart, and it's limited to 100 points. Also, chart updates every 10 seconds instead of every second.

###Version 0.13 (04/29/14)
Even more bugfixes, nerfed clicking upgrades again, added a chart in statistics that tracks money for the session.

###Version 0.12 (04/28/14)
More bugfixes, split money/tick into actual money per tick and money per autoclick, nerfed autoclickers.

###Version 0.11 (04/27/14)
Squashed a lot of bugs, nerfed clicking upgrades, added tier 5 upgrade, modified reset currency gain (in particular mathematician currency gain is clearly visible now). Also, reset, because I screwed up the resetting on the last version.

###Version 0.1 (04/26/14)
Added prestige (plus one unlockable derivative) and Click Improver upgrade, and did some rebalancing. Also implemented dynamic savefiles, so hopefully no more resets will be necessary from this point forward (although balance will probably be tweaked).

###Version 0.031 (03/31/14)
Changes to the number to buy feature: added buy maximum, button lighting reflects number to buy

###Version 0.03 (03/28/14)
Added ability to buy multiple buildings at once, and buffed autoclickers some more.

###Version 0.02 (03/20/14)
Balancing changes (buffed autoclickers, nerfed multipliers, changed prices of some things). Also added upper navigation bar.

###Version 0.01 (03/18/14)
First actual version, and first version I'd call a game.


changelog.txt
Changelog:

Version 0.34 (01/05/2015)
Added infini-currency tracking. Currently, these do nothing.

Version 0.331 (08/17/2014)
Made tooltips less ugly, edited upper section of UI, added achievement number display

Version 0.33 (08/17/2014)
Added base achievement code and a few achievements, added chart options, fixed bug with costs of click booster

Version 0.321 (08/05/2014)
Fixed a bug relating to reset currency buyable display, added number suffixes and link to Github repository

Version 0.32 (08/01/2014)
Add a new option that allows you set the minimum tick length, changed tick length display, fixed clicks to gain going negative, change initial computer price

Version 0.31 (07/28/2014)
Make tooltips not flash

Version 0.3 (07/22/2014)
Added reset currency buyables, which have various effects; players start with $0.10 on reset to allow for neverclick runs

Version 0.222 (07/08/2014)
Number to buy now works with upgrades (except click improver), fixed a minor bug causing money to go negative

Version 0.221 (07/08/2014)
Tier unlocking makes more sense, fixed negative buildings bug

Version 0.22 (07/04/2014)
Added 7th tier, changed reset currency calculation, proper upgrade hiding, improved multiplier calculation, nerfed upgrade multipliers, added alternate number view, changed some names, made some things more efficient, rebalancing.

Version 0.21 (06/27/2014)
Changed mathematician prestige formula to an explicit formula, bugfixes for version 0.2

Version 0.2 (06/27/2014)
Added a sixth derivative (currently unlocked on first 2nd tier reset) as well as related sixth tier stuff, made 5th tier reset possible, refactored some code.

Version 0.184 (06/25/2014)
Changed prestige proof formula to an explicit formula, moved update function into global namespace

Version 0.183 (06/20/2014)
Added number suffixes to trigintillion, added total ticks statistic, added time multiplier (which multiplies everything by 10 when you get to 100 ms tick length, to speed up the game)

Version 0.182 (06/13/2014)
More number display fixes, buttons use onmousedown events for greater responsiveness

Version 0.181 (06/06/2014)
Bugfixes: global multiplier gets recalculated when save imports, numbers that displayed incorrectly have been fixed, some incorrect prices are changed, updating should be a little more responsive

Version 0.18 (05/30/14)
Added tiered resets, and made it way easier to get reset currency.

Version 0.17 (05/16/14)
Fixed some graphical bugs, made chart not show more than 100 points, extended number suffixes to a vigintillion, fixed serious bug due to Chrome and Firefox locking updates to once per second while in the background.

Version 0.16 (05/06/14)
Importing and exporting saves is now possible on the Options screen.

Version 0.15 (05/01/14)
Charts are now more responsive, and there's a button to change the axis.

Version 0.14 (04/30/14)
A lot of changes to try to speed up the chart. Toggling actually disables the chart, and it's limited to 100 points. Also, chart updates every 10 seconds instead of every second.

Version 0.13 (04/29/14)
Even more bugfixes, nerfed clicking upgrades again, added a chart in statistics that tracks money for the session.

Version 0.12 (04/28/14)
More bugfixes, split money/tick into actual money per tick and money per autoclick, nerfed autoclickers.

Version 0.11 (04/27/14)
Squashed a lot of bugs, nerfed clicking upgrades, added tier 5 upgrade, modified reset currency gain (in particular mathematician currency gain is clearly visible now). Also, reset, because I screwed up the resetting on the last version.

Version 0.1 (04/26/14)
Added prestige (plus one unlockable derivative) and Click Improver upgrade, and did some rebalancing. Also implemented dynamic savefiles, so hopefully no more resets will be necessary from this point forward (although balance will probably be tweaked).

Version 0.031 (03/31/14)
Changes to the number to buy feature: added buy maximum, button lighting reflects number to buy

Version 0.03 (03/28/14)
Added ability to buy multiple buildings at once, and buffed autoclickers some more.

Version 0.02 (03/20/14)
Balancing changes (buffed autoclickers, nerfed multipliers, changed prices of some things). Also added upper navigation bar.

Version 0.01 (03/18/14)
First actual version, and first version I'd call a complete game.

js/data.js
var startPlayer = {
	//currencies
	money: 0.1,
	moneyPerSecond: 0,
	netMoneyPerSecond: 0,
	moneyPerClick: 1,
	moneyPerAutoclick: 0,
	clickPower: 1,
	proofs: 0,
	proofsPerSecond: 0,
	costPerProof: 5,
	deriv1Money: 0.05,
	
	/* main buildings: indexes
	0: deriv1, 1: combinatorics, 2: computer, 3: high schooler, 4: mathematician
	5: deriv2, 6: probability, 7: assembly line, 8: undergraduate, 9: andrew wiles
	10: deriv3, 11: number theory, 12: factory, 13: graduate student, 14: kurt godel
	15: deriv4, 16: calculus, 17: factory architect, 18: postdoc, 19: bernhard riemann 
	20: deriv5, 21: algebra, 22: design school, 23: research scientist, 24: carl gauss
	25: deriv6, 26: geometry, 27: dean of architecture, 28: lab manager, 29: leonhard euler
	30: deriv7, 31: arithmetic, 32: chancellor, 33: research lab, 34: isaac newton
	*/
	buildings: [new Building(1.1, 0.1, 0), new Building(1.1, 25000, 0), new Building(1.3, 0, 3), new Building(1.1, 5, 0), new Building(1.05, 1000, 0),
				 new Building(1.2, 500, 0), new Building(1.3, 20000000, 0), new Building(1.8, 0, 20000), new Building(1.2, 1000, 0), new Building(1.2, 100000000, 0),
				 new Building(1.3, 20000, 0), new Building(1.8, 1000000000, 0), new Building(2.5, 0, 1000000), new Building(1.4, 100000, 0), new Building(1.5, 10000000000000, 0),
				 new Building(1.5, 1000000, 1000), new Building(2.5, 500000000000, 0), new Building(4, 0, 30000000), new Building(2, 10000000, 0), new Building(2, 1000000000000000000, 0),
				 new Building(2, 200000000, 500000), new Building(3, 20000000000000, 0), new Building(5, 0, 1000000000), new Building(2.7, 10000000000, 0), new Building(3, 100000000000000000000000, 0),
				 new Building(2.3, 50000000000, 25000000), new Building(3.5, 10000000000000000, 0), new Building(6, 0, 300000000000), new Building(3.2, 10000000000000, 0), new Building(3.5, 10000000000000000000000000, 0),
				 new Building(2.7, 1000000000000, 1000000000), new Building(4, 500000000000000000, 0), new Building(8, 0, 10000000000000), new Building(3.7, 1000000000000000, 0), new Building(4, 1000000000000000000000000000, 0)],

	/* upgrades: indexes
	tier upgrades: 0: tier 1 upgrade, 1: tier 2 upgrade, 2: tier 3 upgrade, 3: tier 4 upgrade, 4: tier 5 upgrade, 5: tier 6 upgrade, 6: tier 7 upgrade
	upgrades: 0: autoclicker, 1: click improver
	*/	
	tierUpgrades: [0, 0, 0, 0, 0, 0, 0],
	upgrades: [0, 0],
	tierUpgradeCosts: [1000000, 1000000000, 1000000000000, 1000000000000000, 1000000000000000000, 1000000000000000000000, 1000000000000000000000000],
	upgradeCosts: [100000, 10000000],
	
	//upgrade multipliers
	mult: [1, 1, 1, 1, 1, 1, 1],
	
	//other tracking variables
	clickTracker: 0,
	updateInterval: 1000,
	numToBuy: 1,
	timeMult: 1,
	proofCurrPerTick: 0,
	mathematicianCurrPerTick: 0,
	
	//these variables aren't changed by resets	
	//statistics
	totalMoneyEarned: 0,
	totalProofs: 0,
	totalClicks: 0,
	totalManualClicks: 0,
	totalTicks: 0,
	proofsToNextCurr: 10000000000000,
	proofsToCurrTracker: 0,
	mathematiciansToNextCurr: 7000000000,
	mathematiciansToNextCurrTracker: 0,
	resetCurrTracker: 0, //this variable does have to be reset
	infiniCurr: 0,
	infiniResetDone: false,
	
	//reset currency buyable things: all reset curr stuff is erased by infini-resets
	resetCurrFactor: 5, //integer
	buildingInterval: 10,
	autoclickInterval: 60,
	clicksToGain: 25,
	
	/*reset currency buyables: columns down
	 * 0: tier 5,                  6: click improver stage 1,  12: building tick decrease stage 1, 18: autoclicker tick decrease stage 1
	 * 1: tier 6,                  7: click improver stage 2,  13: building tick decrease stage 2, 19: autoclicker tick decrease stage 2
	 * 2: tier 7,                  8: click improver stage 3,  14: building tick decrease stage 3, 20: autoclicker tick decrease stage 3
	 * 3: factor decrease stage 1, 9: click improver stage 4,  15: building tick decrease stage 4, 21: reset currency conversion factor stage 1
	 * 4: factor decrease stage 2, 10: click improver stage 5, 16: building tick decrease stage 5, 22: reset currency conversion factor stage 2
	 * 5: factor decrease stage 3, 11: click improver stage 6, 17: building tick decrease stage 6, 23: reset currency conversion factor stage 3
	 */
	currBuyables: [new CurrBuyable([10, 0, 0, 0, 0, 0]), new CurrBuyable([0, 50, 0, 0, 0, 0]), new CurrBuyable([0, 0, 100, 0, 0, 0]), new CurrBuyable([0, 0, 0, 100, 0, 0]), new CurrBuyable([0, 0, 0, 0, 100, 0]), new CurrBuyable([0, 0, 0, 0, 0, 100]),
	               new CurrBuyable([1e3, 0, 0, 0, 0, 0]), new CurrBuyable([0, 1e3, 0, 0, 0, 0]), new CurrBuyable([0, 0, 2e3, 0, 0, 0]), new CurrBuyable([0, 0, 0, 5e3, 0, 0]), new CurrBuyable([0, 0, 0, 0, 1e4, 0]), new CurrBuyable([0, 0, 0, 0, 0, 2e4]),
				   new CurrBuyable([1e5, 0, 0, 0, 0, 0]), new CurrBuyable([0, 1e5, 0, 0, 0, 0]), new CurrBuyable([0, 0, 2e5, 0, 0, 0]), new CurrBuyable([0, 0, 0, 2e5, 0, 0]), new CurrBuyable([0, 0, 0, 0, 5e5, 0]), new CurrBuyable([0, 0, 0, 0, 0, 5e5]),
	               new CurrBuyable([1e7, 0, 0, 0, 0, 0]), new CurrBuyable([0, 1e7, 0, 0, 0, 0]), new CurrBuyable([0, 0, 2e7, 0, 0, 0]), new CurrBuyable([0, 0, 0, 1e9, 0, 0]), new CurrBuyable([0, 0, 0, 0, 2e9, 0]), new CurrBuyable([0, 0, 0, 0, 0, 5e9])],
	
	achievements: [new Achievement("First Click", "Click for the first time!", "player.totalManualClicks > 0"), new Achievement("Clicking 'Expert'", "Click 100 times", "player.totalManualClicks > 100"),
				   new Achievement("Needs More Click", "Click 1000 times", "player.totalManualClicks > 1000"), new Achievement("Algorithmic Clicker", "Click 10000 times", "player.totalManualClicks > 10000"),
				   new Achievement("Clickmeister", "Click 100000 times", "player.totalManualClicks > 100000"), new Achievement("Clickmaster", "Click 1 million times", "player.totalManualClicks > 1e6"),
				   new Achievement("You're Probably Autoclicking", "Click 10 million times", "player.totalManualClicks > 1e7"), new Achievement("It begins...", "Buy a first derivative", "player.buildings[0].manual > 0"),
				   new Achievement("Quadratic Growth", "Buy a second derivative", "player.buildings[5].manual > 0"), new Achievement("Cubic Growth", "Buy a third derivative", "player.buildings[10].manual > 0"),
				   new Achievement("Quartic Growth", "Buy a fourth derivative", "player.buildings[15].manual > 0"), new Achievement("Quintic Growth", "Buy a fifth derivative", "player.buildings[20].manual > 0"),
				   new Achievement("Sextic Growth", "Buy a sixth derivative", "player.buildings[25].manual > 0"), new Achievement("Septic Growth", "Buy a seventh derivative", "player.buildings[30].manual > 0"),
				   new Achievement("Fermat's Last Theorem", "Have an Andrew Wiles", "player.buildings[9].owned > 0"), new Achievement("Incompleteness Theorem", "Have a Kurt Godel", "player.buildings[14].owned > 0"),
				   new Achievement("Riemann Sums", "Have a Bernhard Riemann", "player.buildings[19].owned > 0"), new Achievement("Fundamental Theorem of Algebra", "Have a Carl Gauss", "player.buildings[24].owned > 0"),
				   new Achievement("Euler's Identity", "Have a Leonhard Euler", "player.buildings[29].owned > 0"), new Achievement("Calculus", "Have an Isaac Newton", "player.buildings[34].owned > 0"),
				   new Achievement("Start From Scratch", "Do a tier 1 reset", "player.numResets[0] > 0"), new Achievement("High Tier", "Do a tier 6 reset", "player.numResets[5] > 0"),
				   new Achievement("Millionaire", "Make a million dollars in total", "player.totalMoneyEarned > 1e6"), new Achievement("Billionaire", "Make a billion dollars in total", "player.totalMoneyEarned > 1e9"),
				   new Achievement("...Quadrillionaire?", "Make a quadrillion dollars in total", "player.totalMoneyEarned > 1e15"), new Achievement("Decillionaire", "Make a decillion dollars in total", "player.totalMoneyEarned > 1e33"),
				   new Achievement("Vigintillionaire", "Make a vigintillion dollars in total", "player.totalMoneyEarned > 1e63"), new Achievement("Yeah this isn't a real thing", "Make a trigintillion dollars in total", "player.totalMoneyEarned > 1e93"),
				   new Achievement("Centillionaire", "Make a centillion dollars in total", "player.totalMoneyEarned > 1e303")],
	//settings
	sciNotation: false,
	minTickLength: 100,
	chartDelay: 10000,
	chartLength: 100,
	
	//the next two variables are erased by infini-resets
	numResets: [0, 0, 0, 0, 0, 0, 0],
	resetCurr: [0, 0, 0, 0, 0, 0, 0],
	versionNum: versionNum
};

var resetPlayer = {
	money: 0.1,
	moneyPerSecond: 0,
	netMoneyPerSecond: 0,
	moneyPerClick: 1,
	clickPower: 1,
	moneyPerAutoclick: 0,
	proofs: 0,
	proofsPerSecond: 0,
	costPerProof: 5,
	deriv1Money: 0.05,
	buildings: [new Building(1.1, 0.1, 0), new Building(1.1, 25000, 0), new Building(1.3, 0, 3), new Building(1.1, 5, 0), new Building(1.05, 1000, 0),
				 new Building(1.2, 500, 0), new Building(1.3, 20000000, 0), new Building(1.8, 0, 20000), new Building(1.2, 1000, 0), new Building(1.2, 100000000, 0),
				 new Building(1.3, 20000, 0), new Building(1.8, 1000000000, 0), new Building(2.5, 0, 1000000), new Building(1.4, 100000, 0), new Building(1.5, 10000000000000, 0),
				 new Building(1.5, 1000000, 1000), new Building(2.5, 500000000000, 0), new Building(4, 0, 30000000), new Building(2, 10000000, 0), new Building(2, 1000000000000000000, 0),
				 new Building(2, 200000000, 500000), new Building(3, 20000000000000, 0), new Building(5, 0, 1000000000), new Building(2.7, 10000000000, 0), new Building(3, 100000000000000000000000, 0),
				 new Building(2.3, 50000000000, 25000000), new Building(3.5, 10000000000000000, 0), new Building(6, 0, 300000000000), new Building(3.2, 10000000000000, 0), new Building(3.5, 10000000000000000000000000, 0),
				 new Building(2.7, 1000000000000, 1000000000), new Building(4, 500000000000000000, 0), new Building(8, 0, 10000000000000), new Building(3.7, 1000000000000000, 0), new Building(4, 1000000000000000000000000000, 0)],
	tierUpgrades: [0, 0, 0, 0, 0, 0, 0],
	upgrades: [0, 0],
	tierUpgradeCosts: [1e6, 1e9, 1e12, 1e15, 1e18, 1e21, 1e24],
	upgradeCosts: [100000, 10000000],
	mult: [1, 1, 1, 1, 1, 1, 1],
	clickTracker: 0,
	updateInterval: 1000,
	timeMult: 1,
	resetCurrTracker: 0,
	proofCurrPerTick: 0,
	mathematicianCurrPerTick: 0
}

var infiniResetPlayer = {
	money: 0.1,
	moneyPerSecond: 0,
	netMoneyPerSecond: 0,
	moneyPerClick: 1,
	clickPower: 1,
	moneyPerAutoclick: 0,
	proofs: 0,
	proofsPerSecond: 0,
	costPerProof: 5,
	deriv1Money: 0.05,
	buildings: [new Building(1.1, 0.1, 0), new Building(1.1, 25000, 0), new Building(1.3, 0, 3), new Building(1.1, 5, 0), new Building(1.05, 1000, 0),
				 new Building(1.2, 500, 0), new Building(1.3, 20000000, 0), new Building(1.8, 0, 20000), new Building(1.2, 1000, 0), new Building(1.2, 100000000, 0),
				 new Building(1.3, 20000, 0), new Building(1.8, 1000000000, 0), new Building(2.5, 0, 1000000), new Building(1.4, 100000, 0), new Building(1.5, 10000000000000, 0),
				 new Building(1.5, 1000000, 1000), new Building(2.5, 500000000000, 0), new Building(4, 0, 30000000), new Building(2, 10000000, 0), new Building(2, 1000000000000000000, 0),
				 new Building(2, 200000000, 500000), new Building(3, 20000000000000, 0), new Building(5, 0, 1000000000), new Building(2.7, 10000000000, 0), new Building(3, 100000000000000000000000, 0),
				 new Building(2.3, 50000000000, 25000000), new Building(3.5, 10000000000000000, 0), new Building(6, 0, 300000000000), new Building(3.2, 10000000000000, 0), new Building(3.5, 10000000000000000000000000, 0),
				 new Building(2.7, 1000000000000, 1000000000), new Building(4, 500000000000000000, 0), new Building(8, 0, 10000000000000), new Building(3.7, 1000000000000000, 0), new Building(4, 1000000000000000000000000000, 0)],
	tierUpgrades: [0, 0, 0, 0, 0, 0, 0],
	upgrades: [0, 0],
	tierUpgradeCosts: [1e6, 1e9, 1e12, 1e15, 1e18, 1e21, 1e24],
	upgradeCosts: [100000, 10000000],
	mult: [1, 1, 1, 1, 1, 1, 1],
	clickTracker: 0,
	updateInterval: 1000,
	timeMult: 1,
	proofsToNextCurr: 10000000000000,
	proofsToCurrTracker: 0,
	mathematiciansToNextCurr: 7000000000,
	mathematiciansToNextCurrTracker: 0,
	totalMoneyEarned: 0,
	totalProofs: 0,
	totalClicks: 0,
	totalManualClicks: 0,
	totalTicks: 0,
	resetCurrTracker: 0,
	proofCurrPerTick: 0,
	mathematicianCurrPerTick: 0,
	resetCurrFactor: 5,
	buildingInterval: 10,
	autoclickInterval: 60,
	clicksToGain: 25,
	currBuyables: [new CurrBuyable([10, 0, 0, 0, 0, 0]), new CurrBuyable([0, 50, 0, 0, 0, 0]), new CurrBuyable([0, 0, 100, 0, 0, 0]), new CurrBuyable([0, 0, 0, 100, 0, 0]), new CurrBuyable([0, 0, 0, 0, 100, 0]), new CurrBuyable([0, 0, 0, 0, 0, 100]),
	               new CurrBuyable([1e3, 0, 0, 0, 0, 0]), new CurrBuyable([0, 1e3, 0, 0, 0, 0]), new CurrBuyable([0, 0, 2e3, 0, 0, 0]), new CurrBuyable([0, 0, 0, 5e3, 0, 0]), new CurrBuyable([0, 0, 0, 0, 1e4, 0]), new CurrBuyable([0, 0, 0, 0, 0, 2e4]),
				   new CurrBuyable([1e5, 0, 0, 0, 0, 0]), new CurrBuyable([0, 1e5, 0, 0, 0, 0]), new CurrBuyable([0, 0, 2e5, 0, 0, 0]), new CurrBuyable([0, 0, 0, 2e5, 0, 0]), new CurrBuyable([0, 0, 0, 0, 5e5, 0]), new CurrBuyable([0, 0, 0, 0, 0, 5e5]),
	               new CurrBuyable([1e7, 0, 0, 0, 0, 0]), new CurrBuyable([0, 1e7, 0, 0, 0, 0]), new CurrBuyable([0, 0, 2e7, 0, 0, 0]), new CurrBuyable([0, 0, 0, 1e9, 0, 0]), new CurrBuyable([0, 0, 0, 0, 2e9, 0]), new CurrBuyable([0, 0, 0, 0, 0, 5e9])],
   	numResets: [0, 0, 0, 0, 0, 0, 0],
	resetCurr: [0, 0, 0, 0, 0, 0, 0]
}

var versionNum = 0.341;

//these variables hold constants between plays
var upgradeCostFactor = [1.5, 100];

var globalMult = [1, 1, 1, 1, 1, 1, 1];

var numTiers = 7;